PGIP/Server.hs

```diff
@@ -553,7 +543,15 @@ parseRESTful
               cmdOptList
             newOpts = foldl makeOpts opts $ mapMaybe (`lookup` optionFlags)
               optFlags
-        in if elem newIde newRESTIdes && all (`elem` globalCommands) cmdList
+            validReasoningParams =
+              if elem newIde ["prove", "consistency-check"]
+              then case reasoningParametersE of
+                     Left _ -> False
+                     Right _ -> True
+              else True
```

Comment:
shorter:
`not (validReasoningParams = elem newIde ["prove", "consistency-check"]) || Data.Either.isRight reasoningParametersE `
or even
`Data.Either.isRight reasoningParametersE <= (validReasoningParams = elem newIde ["prove", "consistency-check"])`
Here, `<=` is read as "if", i.e. reverse "implies". See also https://mail.haskell.org/pipermail/libraries/2016-January/026565.html